Crash on nil device in snmp file

When there's a buggy yaml file you can crash ktrans with a nil snmp device. Can argue you should fix your inputs but on general principle its probably better to guard this too. 

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0xb8 pc=0x105be2b08]

goroutine 14 [running]:
github.com/kentik/ktranslate/pkg/inputs/snmp.setDeviceTagsAndMatch(0x0)
	/Users/pye/src/ktranslate/pkg/inputs/snmp/snmp.go:556 +0x28
github.com/kentik/ktranslate/pkg/inputs/snmp.parseConfig({0x1070153f8, 0x14001242150}, {0x16b4f371f, 0x10}, {0x107022930, 0x14000ccc360})
	/Users/pye/src/ktranslate/pkg/inputs/snmp/snmp.go:473 +0xad4
github.com/kentik/ktranslate/pkg/inputs/snmp.initSnmp({0x1070153f8, 0x14001242150}, {0x16b4f371f, 0x10}, {0x107022930, 0x14000ccc360})
	/Users/pye/src/ktranslate/pkg/inputs/snmp/snmp.go:132 +0xbc
github.com/kentik/ktranslate/pkg/inputs/snmp.StartSNMPPolls({0x1070153f8, 0x14001242150}, 0x14000118840, 0x14001300060, {0x10701f9f0, 0x1085d7f60}, 0x0, {0x107022930, 0x14000ccc360}, 0x14000cc50e0, ...)
	/Users/pye/src/ktranslate/pkg/inputs/snmp/snmp.go:65 +0x78
github.com/kentik/ktranslate/pkg/cat.(*KTranslate).Run(0x1400017b520, {0x1070153f8, 0x14001242150})
	/Users/pye/src/ktranslate/pkg/cat/kkc.go:688 +0x6f0
github.com/kentik/ktranslate/pkg/eggs/baseserver.(*BaseServer).Run.func1({0x1070153f8, 0x14001242150})
	/Users/pye/src/ktranslate/pkg/eggs/baseserver/baseserver.go:247 +0x5c
created by github.com/kentik/ktranslate/pkg/eggs/baseserver.(*BaseServer).Run in goroutine 1
	/Users/pye/src/ktranslate/pkg/eggs/baseserver/baseserver.go:245 +0x224
```

Closes https://github.com/kentik/ui-app/issues/23262

--- a/pkg/inputs/snmp/snmp.go
+++ b/pkg/inputs/snmp/snmp.go
@@ -469,7 +469,11 @@ func parseConfig(ctx context.Context, file string, log logger.ContextL) (*kt.Snm
 	}
 
 	// Correctly format all the user tags needed here:
-	for _, device := range ms.Devices {
+	for dname, device := range ms.Devices {
+		if device == nil {
+			log.Errorf("Device %s is nil. Skipping.", dname)
+			continue
+		}
 		setDeviceTagsAndMatch(device) // Tweak any per provider tags and match attributes here now that we have the actual provider.
 		device.InitUserTags(ServiceName)
 	}

